File: backend/src/main/java/az/azcup/backend/BackendApplication.java
package az.azcup.backend;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

// Tətbiqin giriş nöqtəsi (entry point). @SpringBootApplication annotasiyası
// üç şeyi birləşdirir: avtomatik konfiqurasiya (məs. datasource-u application.yml-dən
// oxumaq), komponent-skan (bu paket və alt-paketlərdəki bütün @Service/@Component/
// @RestController siniflərini tapıb qeydiyyatdan keçirmək) və Spring konfiqurasiyası.
@SpringBootApplication
public class BackendApplication {

    // Tətbiq buradan başlayır: daxili Tomcat serverini qaldırır, verilənlər bazasına
    // qoşulur, bütün bean-ları yaradır və CommandLineRunner-ləri (məs.
    // AdminBootstrapRunner, SeedLoader) işə salır.
    public static void main(String[] args) {
        SpringApplication.run(BackendApplication.class, args);
    }
}

File: backend/src/main/kotlin/az/azcup/backend/BackendApplication.kt
package az.azcup.backend

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

// Tətbiqin giriş nöqtəsi (entry point). @SpringBootApplication annotasiyası
// üç şeyi birləşdirir: avtomatik konfiqurasiya (məs. datasource-u application.yml-dən
// oxumaq), komponent-skan (bu paket və alt-paketlərdəki bütün @Service/@Component/
// @RestController siniflərini tapıb qeydiyyatdan keçirmək) və Spring konfiqurasiyası.
@SpringBootApplication
class BackendApplication

// Tətbiq buradan başlayır: daxili Tomcat serverini qaldırır, verilənlər bazasına
// qoşulur, bütün bean-ları yaradır və CommandLineRunner-ləri (məs.
// AdminBootstrapRunner, SeedLoader) işə salır.
fun main(args: Array<String>) {
    runApplication<BackendApplication>(*args)
}
